sympy/ntheory/factor_.py

```diff
@@ -520,11 +520,19 @@ def perfect_power(n, candidates=None, big=True, factor=True):
 
     n = as_int(n)
     if n < 0:
-        pp = perfect_power(-n)
+        pp = perfect_power(-n, candidates=candidates, big=big, factor=factor)
         if pp:
             b, e = pp
-            if e % 2:
-                return -b, e
+            odd_divisors = []
+            for i in range(3, e + 1, 2):
+                if e % i == 0 and i > 1:
+                    odd_divisors.append(i)
+
+            if odd_divisors:
+                chosen_factor = max(odd_divisors) if big else min(odd_divisors)
+                power = e // chosen_factor
+                new_base = -int(b ** power)
+                return new_base, chosen_factor
```

Comment:
pp = perfect_power(-n, candidates, big=True, factor=factor)
        if pp:
            b, e=pp
            e2 = e&(-e)
            e //= e2
            b = b**e2
            if not big and not isprime(e):
                for small in primerange(3, e+1):
                    if not e % small:
                        break
                b = b**(e//small)
                e = small
            return -b, e

Reply:
I think this can be still be more simplified.

Reply:
This works well, especially at handling the suggested test cases in the previous comment. 

> ```python
>    if big or isprime(e):
>        return b, e
>  ```

Maybe here we might have two branches for the return statement.  Basically there must be some more handling here. Directly returning `(b, e)` might be problematic. 

For example, 

```python
>>> perfect_power(-64)
(4, 3) 
# b wasn't converted back to a negative int and the fix isn't as straightforward as 
# returning (-b, e) instead since that breaks down obviously at the test cases in the 
# prev comment

>>> perfect_power(-3**60)
(81, 15) 
# Ideal behavior here 

#But,
>>> perfect_power(-3**61)
(3, 61) # This is a breakdown if I'm not wrong, maybe it should be (-3, 61)
```

I have given a view to this and tried to fix it but the currently my go at the refactoring is pretty naive and ultimately breaks down again.

Reply:
Actually I feel it is as straightforward in this case as returning `(-b, e)` instead. The approach mentioned above is, I guess, enough and very well written to fix the issue we had to begin with and also be accurate given we return `(-b, e)` instead of `(b, e)`.

If we do that, then:

```python
>>> n = -3**60
>>> perfect_power(n)
(-81, 15)
# I feel this is correct, I was probably making a mistake from the start by misreading this as 
# "-3 raised to the exponent 60" but it rather is "negative of 3 raised to exponent 60"

>>> perfect_power((-3)**60)
(3, 60)
# Here it's now, "-3 raised to the exponent 60", I'm sorry for misreading this ealier,
```

Also, we should add a check for the case when during processing `e` evaluates to `1`. 

```python
>>> perfect_power(-4)
(-4, 1) 
# Although by definition of the function e should be > 1 so this should ideally return False
```